### DISCUSSION TITLE
tiff2vips: subsampled images not supported

### DISCUSSION THREAD

**Discussion by gregordr** [libvips/libvips#2437] [Q&A]
## Bug report

**Describe the bug**

When trying to read metadata with sharp I get the following error:
```
tiff2vips: subsampled images not supported
```

I found an issue explaining more about it in https://github.com/libvips/libvips/issues/1328, but not a clear solution. I am using libvips to extract the size of images and create thumbnails of them. Unlike in the linked issue, I am able to provide you with an image that causes such an issue. Ironically, it seems like Github doesn't support this type of file, so I need to give you a [link](https://www.signatureedits.com/wp-admin/admin-ajax.php?action=useyourdrive-download&account_id=114696266120240563741&id=1187V2hye6FnDLUCbf6EJ-dlb2jZysRKR&dl=1&listtoken=12c6c98a53da66da22de94d6435e1ac9). I hope this stays valid for long enough.

**To Reproduce**
Download the image and try using libvips on it. I did it via sharp(image).metadata

**Expected behavior**
Ideally the image should be processed correctly.

**Actual behavior**
The mentioned error is thrown.

**Environment**
(please complete the following information)
 - OS: Windows 10
 - Vips: v8.11.3

**Additional context**
If even with the image you can't add support to libvips, I would like to hear your opinion on what would best work as a substitute.

**Comment by jcupitt** [comment]
Hi @gregordr,

I had a look at your image and it's a CR2. These are Canon camera raw files that look a bit like TIFF, but are not really TIFF. I think this is out of scope for libvips -- you need a specialist raw converter.

I think most people have moved to libraw for this type of image:

https://www.libraw.org/

It's a library-ization of the famous `dcraw` command-line program.

**Reply by daverupp** [reply]
hello @jcupitt: Is there a way, that we can use libvips with the libraw converter?
when I look at --vips-config, we see 
/LIBS/libraw.so.20.0.0 

Is there a way now to call libvips somehow like this:

vips libraw bubu.cr2 x.jpg 

thx for your help.

**Reply by jcupitt** [reply]
That libraw is not from libvips. Perhaps imagemagick can call it?

libvips would need to add a `librawload` operator.

**Comment by kleisauke** [comment]
A dedicated LibRaw loader is now available on the master branch (which will be part of the upcoming 8.18 release).
